src/madsci_data_manager/madsci/data_manager/data_server.py

```diff
@@ -38,29 +67,194 @@ async def root() -> DataManagerDefinition:
         """Return the DataPoint Manager Definition"""
         return data_manager_definition
 
+    def _upload_file_to_minio(
```

Comment:
Could we extract the minio logic from the data client/server into a shared module with methods that are used by both? Avoid some duplicate logic?